### PR TITLE
Implement toRecord function

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2406
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [package]
 org = "ballerina"
 name = "xmldata"
-version = "2.0.1"
+version = "2.0.2"
```
Comment (daneshk) [inline ballerina/Ballerina.toml:4]: Let's do minor version bump as we are adding new feature

File: ballerina/tests/xml_to_record_test.bal [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+        }
+    };
+
+    BookStore|Error actual = toRecord(payload);
+    if actual is Error {
+        test:assertFail("failed to convert xml to record: " + actual.message());
+    } else {
+        test:assertEquals(actual, expected, msg = "testToRecordWithNamespaces result incorrect");
+    }
+}
```
Comment (daneshk) [inline ballerina/tests/xml_to_record_test.bal:566]: Shall we add some negative testcases? where record type is not match with the XML value

Comment (MadhukaHarith92) [reply ballerina/tests/xml_to_record_test.bal:566]: Added in 5422e3f

File: ballerina/Ballerina.toml (hunk 1 above)
Comment (MadhukaHarith92) [reply ballerina/Ballerina.toml]: Bumped to `2.1.0`

**Review by daneshk — APPROVED** [review]

**Comment by daneshk** [comment]
@kalaiyarasiganeshalingam @MadhukaHarith92 please include this in the specification and proposal

**Comment by daneshk** [comment]
@MadhukaHarith92 We changed the API as below, slightly deferred from the original design. removed `attributePrefix` as we are not using it when converting to record.

```
public isolated function toRecord(xml xmlValue, boolean preserveNamespaces = true, typedesc<record {}> returnType = <>) returns returnType|Error;
```

**Pull request merged**